### PR TITLE
Migrate compiler-plugin tests to new test framework

### PR DESCRIPTION
**Pull request by ting-yuan**

### CODE REVIEW

File: test-utils/src/main/kotlin/com/google/devtools/ksp/testutils/AbstractKSPTest.kt [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    fun runTest(@TestDataFile path: String) {
+        val testConfiguration = testConfiguration(path, configure)
+        Disposer.register(disposable, testConfiguration.rootDisposable)
+        val testServices = testConfiguration.testServices
+        val moduleStructure = testConfiguration.moduleStructureExtractor.splitTestDataByModules(
+            path,
+            testConfiguration.directives,
+        )
+        val mainModule = moduleStructure.modules.last()
+        val libModules = moduleStructure.modules.filterNot { it == mainModule }
```
Comment (neetopia) [inline test-utils/src/main/kotlin/com/google/devtools/ksp/testutils/AbstractKSPTest.kt:196]: nit: if the last module is always the main module, maybe just `dropLast(1)` ?

**Review by neetopia — APPROVED** [review]
LGTM, see minor comments.

File: compiler-plugin/src/test/kotlin/com/google/devtools/ksp/test/KSPTest.kt [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+import org.jetbrains.kotlin.test.services.javaFiles
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+import java.io.File
+
+@Execution(ExecutionMode.SAME_THREAD)
+class KSPTest : AbstractKSPTest(FrontendKinds.ClassicFrontend) {
+    override fun runTest(testServices: TestServices, mainModule: TestModule, libModules: List<TestModule>) {
```
[the remark this answers is not in the document]
Comment (ting-yuan) [reply compiler-plugin/src/test/kotlin/com/google/devtools/ksp/test/KSPTest.kt]: Done

File: test-utils/src/main/kotlin/com/google/devtools/ksp/testutils/AbstractKSPTest.kt (hunk 1 above)
Comment (ting-yuan) [reply test-utils/src/main/kotlin/com/google/devtools/ksp/testutils/AbstractKSPTest.kt]: Done

**Pull request merged**